letsencrypt-win-simple/Plugin/SanByDomainPlugin.cs

```diff
@@ -0,0 +1,288 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Serilog;
+using Microsoft.Web.Administration;
+
+namespace LetsEncrypt.ACME.Simple
+{
+    public class SanByDomainPlugin : Plugin
+    {
+        public override string Name => "SanByDomain";
+        private const int SanMax = 100;
+        //This plugin is designed to allow a user to select one domain and all subdomains for a single San certificate across several sites.
+
+        public override List<Target> GetTargets()
+        {
+            var result = new List<Target>();
+
+            return result;
+        }
+
+        public override List<Target> GetSites()
+        {
+            var result = new List<Target>();
+
+            return result;
+        }
+
+        public override void Install(Target target, string pfxFilename, X509Store store, X509Certificate2 certificate)
+        {
+            Console.WriteLine(" WARNING: Unable to configure server software.");
+        }
+
+        public override void Install(Target target)
+        {
+            Console.WriteLine(" WARNING: Unable to configure server software.");
+        }
+
+        public override void PrintMenu()
+        {
+            Console.WriteLine(" D: Generate a single San certificate by Domain with subdomains.");
+        }
+
+        public override void HandleMenuResponse(string response, List<Target> targets)
+        {
+            if (response == "d")
+            {
+                Console.WriteLine("Running SanByDomain Plugin");
+                Log.Information("Running SanByDomain Plugin");
+
+                List<Domain> domains = new List<Domain>();
+
+                using (var iisManager = new ServerManager())
```

Comment:
You need to instantiate a new plugin object and call the IISPlugin's GetSites to get all of the sites. That will filter out all of the invalid sites.